src/python/pants/engine/internals/rule_visitor_test.py

```diff
@@ -70,6 +80,18 @@ def assert_awaitables(func, awaitable_types: Iterable[tuple[OutT, InT | list[InT
     assert actual_types == expected_types
 
 
+def assert_byname_awaitables(func, awaitable_types: Iterable[tuple[OutT, InT | list[InT], int]]):
```

Comment:
This is the same as `assert_awaitables()` above, but with a third tuple member, representing the explicit_args_arity. 

It didn't seem worth it to try and merge the two. For one thing, the type signature would be hideous.